client/src/utils/apiHelpers.js

```diff
@@ -19,10 +19,14 @@ async function callApi(url, { method, body }) {
     fetchOptions.body = JSON.stringify(body);
   }
   const res = await fetch(`/api/${url}`, fetchOptions);
-  const json = await res.json();
+  let json = {};
+
+  if (res.status !== 204) {
+    json = await res.json();
+  }
 
   if (!res.ok) {
-    const errMessage = json.error ? json.error.message : res.statusText;
+    const errMessage = json && json.error ? json.error.message : res.statusText;
```

Comment:
If we are setting `json` to always be an object, then we don't need this null check, and can just use: `json.error ? ...`